### ISSUE TITLE
useContract and useReadContract are not typesafe on v3

### ISSUE THREAD

**Issue by kseikyo** [apibara/starknet-react#469]
[useContract](https://starknet-react-website.vercel.app/docs/hooks/use-contract#usage) and [useReadContract](https://starknet-react-website.vercel.app/docs/hooks/use-contract#usage) are returning any for `contract` and `data` using the examples from the docs.

Versions on package.json
```json
"starknet": "^6.11.0",
"@starknet-react/chains": "3.0.0-beta.1",
"@starknet-react/core": "3.0.0-beta.1"
```

Link to reproduction:
https://stackblitz.com/edit/stackblitz-starters-uzoe6u?file=app%2Fpage.tsx

**Comment by fracek** [comment]
Thank you for reporting this. Is it confirmed it's a bug in starknet.js?

**Comment by ShantelPeters** [comment]
@kseikyo  please can I be assigned to this ?

**Comment by kseikyo** [comment]
> Thank you for reporting this. Is it confirmed it's a bug in starknet.js?

Its a bug in the abi-wan repo that the declaration merging doesn't work, there's an [issue](https://github.com/keep-starknet-strange/abi-wan-kanabi/issues/72) and [PR](https://github.com/keep-starknet-strange/abi-wan-kanabi/pull/73) opened that fixes it.

Closing for now.

**Comment by mohammedi-haroune** [comment]
Hello @kseikyo,

I just merged the [PR](https://github.com/keep-starknet-strange/abi-wan-kanabi/pull/73) and published version 2.2.3 ton npm, would you please check if this fixes this issue, if not, don't hesitate to reach out to me, I'll be more than happy to help
